### DISCUSSION TITLE
Expo Plugin for Analytics package - Enable AdSupport.framework

### DISCUSSION THREAD

**Discussion by zholmes1** [invertase/react-native-firebase#7789] [Ideas] [answered]
Hey! Just want to start by saying thank you so much for this amazing package. I've built 3 successful app businesses that all make heavy use of RNFB 💛

My most recent app is using Expo, fully managed, no prebuild. I'd like to enable the AdSupport.Framework on iOS, but it seems like I'd need access to native files (Podfile) to make the change described here: https://rnfirebase.io/analytics/usage#device-identification

Any help or guidance here would be appreciated!

**Comment by mike24dzy** [comment]
Hey I am facing the same question. How are we able to enable the AdSupport.Framework for RNFB with fully Expo managed workflow on iOS?

**Comment by zholmes1** [comment, marked as answer]
@mike24dzy 

Add `"./plugins/withPodfile",` to your `expo.plugins` array in `app.json`

/plugins/withPodfile.js

```
const { withDangerousMod } = require('@expo/config-plugins')
const fs = require('fs')
const path = require('path')

const PREPEND_LINE = '$RNFirebaseAnalyticsEnableAdSupport = true'

/**
 *
 * @param {import('expo/config-plugins').ExportedConfig} config
 * @returns {import('expo/config-plugins').ExportedConfigWithProps}
 */
module.exports = config => {
  return withDangerousMod(config, [
    'ios',
    config => {
      const file = path.join(
        config.modRequest.platformProjectRoot,
        'Podfile'
      )

      const contents = fs
        .readFileSync(file, 'utf-8')
        .replace(PREPEND_LINE + '\n\n', '')

      fs.writeFileSync(file, `${PREPEND_LINE}\n\n${contents}`)

      return config
    }
  ])
}
```

**Reply by mike24dzy** [reply]
Thank you very much!!

**Comment by drtapha** [comment]
I added it but the analytics still doesn't work on iOS ? Do you have any ideas ?
